dimscord/helpers/guild.nim

```diff
@@ -0,0 +1,253 @@
+template prune*(g: Guild;
+        days: range[1..30] = 7;
+        include_roles: seq[string] = @[];
+        compute_prune_count = true): Future[void] =
+    ## Begins a guild prune.
+    getClient.api.beginGuildPrune(g.id, days, include_roles, compute_prune_count)
+
+template pruneCount*(g: Guild, days: int): Future[int] =
+    ## Gets the prune count.
+    getClient.api.getGuildPruneCount(g.id, days)
+
+template edit*(g: Guild, lvl: MFALevel): Future[MFALevel] =
+    ## Modify Guild MFA Level, requiring guild ownership.
+    getClient.api.editGuildMFALevel(g.id, lvl)
+
+template delete*(g: Guild): Future[void] =
+    ## Deletes a guild. Requires guild ownership.
+    getClient.api.deleteGuild(g.id)
+
+template edit*(g: Guild;
+        name, description, region, afk_channel_id, icon = none string;
+        discovery_splash, owner_id, splash, banner = none string;
+        system_channel_id, rules_channel_id = none string;
+        preferred_locale, public_updates_channel_id = none string;
+        verification_level, default_message_notifications = none int;
+        system_channel_flags = none int;
+        explicit_content_filter, afk_timeout = none int;
+        features: seq[string] = @[];
+        premium_progress_bar_enabled = none bool;
+        reason = ""
+): Future[Guild] =
+    ## Modifies a guild.
+    ## Icon needs to be a base64 image.
+    ## (See: https://nim-lang.org/docs/base64.html)
+    ## 
+    ## 
+    ## Read more at: 
+    ## https://discord.com/developers/docs/resources/guild#modify-guild
+    getClient.api.editGuild(    
+        g.id, name, description, region, afk_channel_id, icon,
+        discovery_splash, owner_id, splash, banner,
+        system_channel_id, rules_channel_id,
+        preferred_locale, public_updates_channel_id,
+        verification_level, default_message_notifications,
+        system_channel_flags,
+        explicit_content_filter, afk_timeout,
+        features, premium_progress_bar_enabled
+    )
+
+template audits*(g: Guild; 
+        user_id, before = "", action_type = -1;
+        limit: range[1..100] = 50
+): Future[AuditLog] =
+    ## Get guild audit logs. The maximum limit is 100.
+    getClient.api.getGuildAuditLogs(g.id, user_id, before, action_type, limit)
+
+# (!): requires PR to add `guild_id` field
+# template delete*(r: Role): Future[void] =
```

Comment:
for now you can just rename it to deleteRole since Guild is the parent parameter of the endpoint.
```suggestion
# template deleteGuildRole*(g: Guild, r: Role): Future[void] =
```

Reply:
`delete` seems fine no? It already knows its deleting a role from the parameter type so no need to make it verbose

Reply:
I've named it `deleteRole` in the commits after, I will add Role.delete in the next release.